run_dir/static/js/project_samples.js

```diff
@@ -352,12 +375,19 @@ function load_all_udfs(){
         }
       }
 
-      // Make the project emails clickable, and add labels.
+      // Make the project emails clickable and add labels.
       else if (prettify(key) == 'contact'){
         function elabel(text, label) {
           return '<span class="label label-'+label+'">'+text+'</span>'
         }
-        if(data['order_details']) {
+        // Stolen, without shame, from https://stackoverflow.com/a/46181
+        function validateEmail(email) {
+          var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+          return re.test(String(email).toLowerCase());
+        }
+
+        var email_html = '';
+        try {
           var emails = {}
           var contact = data['order_details']['owner']['email'];
```

Comment:
Shouldn't we go for `data['order_details']['fields']['project_lab_email']` first as **Contact** and if not avilable fall back to *owner*. I think *"project_lab_email"* is the API equivalent of *"primary_contact_email"* in the orderportal page.

Reply:
I guess this is the logic that the project coordinators use. I haven't really done any comparisons if this is true that `data['order_details']['fields']['project_lab_email'] ~= data['primary_contact_email']`

But yeah. I think I've seen that the contact person might change when going for the "fallback" option

Reply:
I think I'll leave this as is. I think we should rather think about what `contact person` means than try to over-engineer this.